app/models/opentofu_worker.rb

```diff
@@ -0,0 +1,59 @@
+class OpentofuWorker < MiqWorker
+  self.required_roles = ["embedded_terraform"]
+  self.rails_worker   = false
+
+  def self.service_base_name
+    "opentofu-runner"
+  end
+
+  def self.service_file
+    "#{service_base_name}.service"
+  end
+
+  def self.worker_deployment_name
+    "opentofu-runner"
+  end
+
+  def self.kill_priority
+    MiqWorkerType::KILL_PRIORITY_GENERIC_WORKERS
+  end
+
+  self.maximum_workers_count = 1
+
+  # There can only be a single instance running so the unit name can just be
+  # "opentofu-runner.service"
+  def unit_instance
+    ""
+  end
+
+  def container_image_name
+    "opentofu-runner"
+  end
+
+  def container_image
+    "opentofu-runner"
+  end
+
+  def unit_config_file
+    # Override this in a sub-class if the specific instance needs
+    # any additional config
+    <<~UNIT_CONFIG_FILE
+      [Service]
+      MemoryHigh=#{worker_settings[:memory_threshold].bytes}
+      TimeoutStartSec=#{worker_settings[:starting_timeout]}
+      TimeoutStopSec=#{worker_settings[:stopping_timeout]}
+      #{unit_environment_variables.map { |env_var| "Environment=#{env_var}" }.join("\n")}
+    UNIT_CONFIG_FILE
+  end
```

Comment:
NOTE this is temporary, I had to override this to drop the `WatchdogSec` setting since we're not calling `sd_notify` in this example.
A better way of disabling one of these would be to check if the `worker_settings` was `nil` before adding it to the config_file string.

Reply:
https://github.com/ManageIQ/manageiq/pull/22981